### ISSUE TITLE
Bug: [prefer-optional-chain] does not report for property on intersection type

### ISSUE THREAD

**Issue by ulrichstark** [typescript-eslint/typescript-eslint#12122]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.3&fileType=.tsx&code=C4TwDgpgBAQgTgQwHYGMAWUC8UDeUCWAJgFxQDOwc%2BSA5lAL5QBkuUSCAthKRVbQwG4AUEMIQUAGwRxoKAPZIKUBKXjJ0UAD5sArhInCxk6bIVKARqsSo0wofMXAoKLFACECLdvMA6Iu8xsBD9CASA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uii9o-aJHBgAviHVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type Branch = { id: string } & { name: string };

declare const a: Branch | null;
declare const b: Branch;

const c = !a || b.id !== a.id;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-optional-chain": "error"
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

`!a || b.id !== a.id` should be reported and fixed to `b.id !== a?.id`

### Actual Result

No diagnostics

### Additional Info

Removing `& { name: string }` from the `Branch` type makes the code report.
